bluecellulab/tools.py

```diff
@@ -507,20 +506,74 @@ def validate_section_and_segment(cell: Cell, section_name: str, segment_position
         raise ValueError(f"Segment position must be between 0.0 and 1.0, got {segment_position}.")
 
 
-def get_section(cell: Cell, section_name: str) -> NeuronSection:
-    """Retrieve a NEURON section from the cell by its name.
+def get_section(cell, section_name: str):
+    """Return a single, fully specified NEURON section (e.g., 'soma[0]',
+    'dend[3]').
 
-    Args:
-        cell (Cell): The cell object containing the sections.
-        section_name (str): The name of the section to retrieve.
+    Raises:
+        ValueError or TypeError if the section is not found or invalid.
+    """
+    if section_name in cell.sections:
+        section = cell.sections[section_name]
+        if hasattr(section, "nseg"):
+            return section
+        raise TypeError(f"'{section_name}' exists but is not a NEURON section.")
 
-    Returns:
-        NeuronSection: The NEURON section corresponding to the provided name.
+    available = ", ".join(cell.sections.keys())
+    raise ValueError(f"Section '{section_name}' not found. Available: [{available}]")
+
+
+def get_sections(cell, section_name: str):
+    """Return a list of NEURON sections.
+
+    If the section name is a fully specified one (e.g., 'dend[3]'), return it as a list of one.
+    If the section name is a base name (e.g., 'dend'), return all matching sections like 'dend[0]', 'dend[1]', etc.
 
     Raises:
-        ValueError: If the section with the specified name does not exist.
+        ValueError or TypeError if no valid sections are found.
     """
+    # Try to interpret as fully qualified section name
     try:
-        return cell.sections[section_name]
-    except KeyError:
-        raise ValueError(f"Section '{section_name}' not found in the cell.")
+        return [get_section(cell, section_name)]
+    except ValueError:
+        pass  # Not a precise match; try prefix match
+
+    # Fallback to prefix-based match (e.g., 'dend' → 'dend[0]', 'dend[1]', ...)
+    matched = [
+        section for name, section in cell.sections.items()
+        if name.startswith(f"{section_name}[")
+    ]
+    if matched:
+        return matched
+
+    available = ", ".join(cell.sections.keys())
+    raise ValueError(f"Section '{section_name}' not found. Available: [{available}]")
+
+
+def resolve_segments(cell, report_cfg, node_id, compartment_nodes, source_type):
+    """Determine which segments to record from one or more NEURON sections."""
+    section_name = report_cfg.get("sections", "soma")
+    compartment = report_cfg.get("compartments", "center")
+
+    if source_type == "compartment_set":
+        return [
+            (get_sections(cell, sec)[0], sec, seg)
```

Comment:
Do we expect sec to always be fully specified here? If not, why keeping only the first element in list?